x-pack/platform/plugins/shared/fleet/server/saved_objects/index.ts

```diff
@@ -1258,6 +1267,10 @@ export const OUTPUT_ENCRYPTED_FIELDS = new Set([
   { key: 'password', dangerouslyExposeValue: true },
 ]);
 
+export const FLEET_SERVER_HOST_ENCRYPTED_FIELDS = new Set([
+  { key: 'ssl', dangerouslyExposeValue: true },
```

Comment:
__question:__ Why do we need `dangerouslyExposeValue: true`? Do we have to expose this value to our users in unencrypted form? If so, is it a requirement?

If we decide to keep this property, it'd be helpful to include a code comment with justification.

Reply:
I looked at the use of`getDecryptedAsInternalUser` but at the moment it's not really feasible as it would mean rewriting the service in a totally different way. 

I know that this solution is not ideal, but it's a temporary one and in fact this PR is also adding `secrets.ssl` field that has the same capability and can store those values in a secure way by using secrets. The plan is to remove the use of the encrypted SO field once that's enabled. The only reason why is not already available is that we still need to add the secret reading capabilities to fleet server, but it is absolutely a temporary solution.
cc @kpollich

Reply:
Great, thanks for looking into this and clarifying it for me. Having this as a temporary solution sounds reasonable to me.